demo/apisix/tmail-apisix-plugin-runner/src/test/java/com/linagora/apisix/plugin/ClusterRedisRevokedTokenRepositoryTest.java

```diff
@@ -0,0 +1,72 @@
+package com.linagora.apisix.plugin;
+
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+
+public class ClusterRedisRevokedTokenRepositoryTest implements RevokedTokenRepositoryContract {
+
+    static Network dockernetwork = Network.newNetwork();
+
+    static GenericContainer<?> REDIS_MASTER = new GenericContainer<>(DockerImageName.parse("bitnami/redis:7.0.4-debian-11-r25"))
+        .withNetworkAliases("redis-master")
+        .withCommand()
+        .withNetwork(dockernetwork)
+        .withEnv("REDIS_REPLICATION_MODE", "master")
+        .withEnv("REDIS_PASSWORD", "my_master_password")
+        .withExposedPorts(6379)
+        .withCreateContainerCmdModifier(createContainerCmd -> createContainerCmd.withName("redis-master-test" + UUID.randomUUID()));
+
+    static GenericContainer<?> REDIS_REPLICA = new GenericContainer<>(DockerImageName.parse("bitnami/redis:7.0.4-debian-11-r25"))
+        .withNetwork(dockernetwork)
+        .withNetworkAliases("redis-slave")
+        .withEnv("REDIS_REPLICATION_MODE", "slave")
+        .withEnv("REDIS_MASTER_HOST", "redis-master")
+        .withEnv("REDIS_MASTER_PORT_NUMBER", "6379")
+        .withEnv("REDIS_MASTER_PASSWORD", "my_master_password")
+        .withEnv("REDIS_PASSWORD", "my_replica_password")
+        .withExposedPorts(6379)
+        .dependsOn(REDIS_MASTER)
+        .withCreateContainerCmdModifier(createContainerCmd -> createContainerCmd.withName("redis-replica-test" + UUID.randomUUID()));
+
+    @BeforeAll
+    static void beforeAll() {
+        REDIS_MASTER.start();
+        REDIS_REPLICA.start();
+
+    }
+
+    @AfterAll
+    static void afterAll() {
+        REDIS_MASTER.stop();
+        REDIS_REPLICA.stop();
+    }
+
+    RedisRevokedTokenRepository revokedTokenRepository;
+
+    @BeforeEach
+    void setup() {
+        revokedTokenRepository = new RedisRevokedTokenRepository(AppConfiguration.initRedisCommand(
+            String.format("redis://%s@localhost:%d",
+                "my_master_password", REDIS_MASTER.getMappedPort(6379)), false));
```

Comment:
This testes success now. 
But fail if the input is list uri (master & slave). 

I'm stuck here, I guess maybe by Redis configure not correct somewhere